provider/cluster/kube/builder/workload.go

```diff
@@ -0,0 +1,180 @@
+package builder
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/tendermint/tendermint/libs/log"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	manifesttypes "github.com/ovrclk/akash/manifest"
+	clusterUtil "github.com/ovrclk/akash/provider/cluster/util"
+	mtypes "github.com/ovrclk/akash/x/market/types"
+)
+
+const (
+	StorageAttributePersistent = "persistent"
+	StorageAttributeClass      = "class"
+	StorageAttributeMount      = "mount"
+	StorageAttributeReadOnly   = "readOnly"
+)
+
+type workloadBase interface {
+	builderBase
+	Name() string
+}
+
+type workload struct {
+	builder
+	service          *manifesttypes.Service
+	runtimeClassName string
+}
+
+var _ workloadBase = (*workload)(nil)
+
+func newWorkloadBuilder(log log.Logger, settings Settings, lid mtypes.LeaseID, group *manifesttypes.Group, service *manifesttypes.Service) workload {
+	return workload{
+		builder: builder{
+			settings: settings,
+			log:      log.With("module", "kube-builder"),
+			lid:      lid,
+			group:    group,
+		},
+		service:          service,
+		runtimeClassName: settings.DeploymentRuntimeClass,
+	}
+}
+
+func (b *workload) container() corev1.Container {
+	falseValue := false
+
+	kcontainer := corev1.Container{
+		Name:    b.service.Name,
+		Image:   b.service.Image,
+		Command: b.service.Command,
+		Args:    b.service.Args,
+		Resources: corev1.ResourceRequirements{
+			Limits:   make(corev1.ResourceList),
+			Requests: make(corev1.ResourceList),
+		},
+		ImagePullPolicy: corev1.PullIfNotPresent,
+		SecurityContext: &corev1.SecurityContext{
+			RunAsNonRoot:             &falseValue,
+			Privileged:               &falseValue,
+			AllowPrivilegeEscalation: &falseValue,
+		},
+	}
+
+	if cpu := b.service.Resources.CPU; cpu != nil {
```

Comment:
I know this is basically identical to what is on master, but do we actually have a case where this can be `nil`? Same question applies to memory below this. Thoughts @boz  ?

Reply:
yeah, good point. even with gpu only workloads cpu is required